docs/howto/csrf-protection.txt

```diff
@@ -0,0 +1,321 @@
+
+    function csrfSafeMethod(method) {
+        // these HTTP methods do not require CSRF protection
+        return (/^(GET|HEAD|OPTIONS|TRACE)$/.test(method));
+    }
+    $.ajaxSetup({
+        beforeSend: function(xhr, settings) {
+            if (!csrfSafeMethod(settings.type) && !this.crossDomain) {
+                xhr.setRequestHeader("X-CSRFToken", csrftoken);
+            }
+        }
+    });
+
+If you're using AngularJS 1.1.3 and newer, it's sufficient to configure the
+``$http`` provider with the cookie and header names:
+
+.. code-block:: javascript
+
+    $httpProvider.defaults.xsrfCookieName = 'csrftoken';
+    $httpProvider.defaults.xsrfHeaderName = 'X-CSRFToken';
+
+Using CSRF protection in Jinja2 templates
+=========================================
+
+Django's :class:`~django.template.backends.jinja2.Jinja2` template backend
+adds ``{{ csrf_input }}`` to the context of all templates which is equivalent
+to ``{% csrf_token %}`` in the Django template language. For example:
+
+.. code-block:: html+jinja
+
+    <form action="" method="post">{{ csrf_input }}
+
+Using the decorator method
+==========================
+
+Rather than adding ``CsrfViewMiddleware`` as a blanket protection, you can use
+the :func:`~django.views.decorators.csrf.csrf_protect` decorator, which has
+exactly the same functionality, on particular views that need the protection.
+It must be used **both** on views that insert the CSRF token in the output, and
+on those that accept the POST form data. (These are often the same view
+function, but not always).
+
+Use of the decorator by itself is **not recommended**, since if you forget to
+use it, you will have a security hole. The 'belt and braces' strategy of using
+both is fine, and will incur minimal overhead.
+
+.. _csrf-rejected-requests:
+
+Handling rejected requests
+==========================
+
+By default, a '403 Forbidden' response is sent to the user if an incoming
+request fails the checks performed by ``CsrfViewMiddleware``.  This should
+usually only be seen when there is a genuine Cross Site Request Forgery, or
+when, due to a programming error, the CSRF token has not been included with a
+POST form.
+
+The error page, however, is not very friendly, so you may want to provide your
+own view for handling this condition.  To do this, simply set the
+:setting:`CSRF_FAILURE_VIEW` setting.
+
+CSRF failures are logged as warnings to the :ref:`django.security.csrf
+<django-security-logger>` logger.
+
+.. versionchanged:: 1.11
+
+    In older versions, CSRF failures are logged to the ``django.request``
+    logger.
+
+Using CSRF protection with caching
+==================================
+
+If the :ttag:`csrf_token` template tag is used by a template (or the
+``get_token`` function is called some other way), ``CsrfViewMiddleware`` will
+add a cookie and a ``Vary: Cookie`` header to the response. This means that the
+middleware will play well with the cache middleware if it is used as instructed
```

Comment:
`well` is an overstatement, it doesn't introduce any new security issues but makes caching effectively useless since you'd cache per browser session.

Reply:
Also unchanged, but noted.